### DISCUSSION TITLE
running command with `:! <COMMAND>` doesn't print stderr output

### DISCUSSION THREAD

**Discussion by pmnoxx** [JetBrains/ideavim#434] [Q&A]
I noticed that whenever try to run a local command: ":! something" only stdout gets printed, while stderr gets dropped silently.
Is there any workaround this?

**Comment by citizenmatt** [comment]
There isn't a workaround, but what does normal Vim do here? I know that if the process fails, we show error output, but I can't remember what Vim does

**Reply by pmnoxx** [reply]
> There isn't a workaround, but what does normal Vim do here? I know that if the process fails, we show error output, but I can't remember what Vim does

I'm running my tools, which print errors/warnings to standard error.  When I use those commands from nvim/vim, both stdout and stderr get printed.

**Reply by pmnoxx** [reply]
@citizenmatt Btw, it looks like enviroment variables don't get passed either. Which can be seen by looking at

`:!env`.

**Reply by pmnoxx** [reply]
https://github.com/JetBrains/ideavim/pull/435
